src/ngx_http_lua_kong_module.c

```diff
@@ -670,5 +683,197 @@ ngx_http_lua_kong_get_upstream_ssl_verify(ngx_http_request_t *r,
 
     return ctx->upstream_ssl_verify;
 }
+# endif
+
+char *
+ngx_http_lua_kong_load_var_index(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+{
+
+    ngx_str_t                     *value;
+    ngx_int_t                      index;
+
+    value = cf->args->elts;
+
+    if (value[1].len == 0) {
```

Comment:
If we make the argument starts with `$`, then the following logic can be copied directly from `set` directive:

http://lxr.nginx.org/source/xref/nginx/src/http/modules/ngx_http_rewrite_module.c?r=7606%3Adb8df9cd84c8#908

In that case, the `value[1].len == 0` check is not necessary. L908-L915 is probably enough here.

Reply:
done